adaptivefiltering/segmentation.py

```diff
@@ -54,9 +59,271 @@ def save(self, filename):
         with open(filename, "w") as f:
             geojson.dump(self, f)
 
+    def show(self):
+        """This will create a new InteractiveMap with bounds from the segmentation.
+             use
+                 map = self.show()
+                 map
+             to show and access the data of the new map or just
+                self.show() to show the map without interacting with it.
+        :param grid:
+             The grid object which holds the map and the right side interface
+         :type grid: ipyleaflet.grid
+        """
+
+        segmentation_map = InteractiveMap(segmentation=self)
+        return segmentation_map.show()
+
     @property
     def __geo_interface__(self):
         return {
             "type": "FeatureCollection",
             "features": self.features,
         }
+
+
+class InteractiveMap:
+    def __init__(self, dataset=None, segmentation=None):
+        """This class manages the interactive map on which one can choose the segmentation.
+            It can be initilized with a dataset from which it will detect the boundaries and show them on the map.
+            Or it can be initilized with a segmentation which will also be visualized on the map.
+            There can be multiple polygons in the Segmentation and all will drawn. The
+
+
+        :param dataset:
+            The dataset from which the map should be displayed. This needs to be in a valid georeferenced format. Eg.: EPSG:4326
+        :type dataset: Dataset
+        :param segmentation:
+            A premade segmentation can can be loaded and shown on a map without the need to load a dataset.
+        :type segmentation: Segmentation
+
+        """
+
+        # handle exeptions
+
+        if dataset and segmentation:
+            raise Exception(
+                "A dataset and a segmentation can't be loaded at the same time."
+            )
+
+        if dataset is None and segmentation["features"] == []:
+            raise Exception("an empty segmention was given.")
+
+        if dataset is None and segmentation is None:
+            # if no dataset or segmentation is given, the map will be centered at the SSC office
+            self.coordinates_mean = np.asarray([49.41745, 8.67529])
+            self.segmentation = None
+
+        if dataset is not None and type(dataset) is not DataSet:
```

Comment:
`type(x) is Foo` should generally be avoided in favor of `isinstance(x, Foo)` to allow subclassing by users. It is more pythonic.